### ISSUE TITLE
TS: SQLDatabase's connection string fails when connecting to Prisma CLI

### ISSUE THREAD

**Issue by luisnquin** [encoredev/encore#1388]
At my organization we're trying to integrate **Prisma** with **Encore.dev(TS)**. Fow now it looks like [Prisma doesn't allows you to programatically migrate everything](https://github.com/prisma/prisma/issues/4703) so we decided to spawn the Prisma CLI as a workaround.

The problem here is that the _connection string_ provided by `SQLDatabase` is wrong but the Encore CLI is able to give me the correct connection string.

![encore-cli-conn-uri](https://github.com/user-attachments/assets/e5e0aa65-ba0c-40c6-b419-2f293d5472da)

## Code

```ts
// auth/auth.ts
import { SQLDatabase } from "encore.dev/storage/sqldb";
import { prismaMigrate } from "@/lib/prisma";

const db = new SQLDatabase("auth");

console.log("[1] connection string is...", db.connectionString);

const result = await db.queryRow`SELECT 1`;
console.log("result is...", result);

console.log("[2] connection string is...", db.connectionString);

await prismaMigrate(db.connectionString);

```

```ts
// lib/prisma.ts
// Code from https://github.com/prisma/prisma/issues/4703#issuecomment-1447354363
import { promisify } from "node:util";
import { exec as execCb } from "node:child_process";

const exec = promisify(execCb);

// TODO: re-write this when Prisma.io gets a programmatic migration API
// https://github.com/prisma/prisma/issues/4703
export async function prismaMigrate(databaseUrl: string): Promise<void> {
  // throws an error if migration fails
  const { stdout, stderr } = await exec("npx prisma db push", {
    env: {
      ...process.env,
      DATABASE_URL: databaseUrl,
    },
  });
  console.log(stdout);
  console.warn(stderr);
}

```

## Terminal

| ![local-server-logs](https://github.com/user-attachments/assets/0b9691c4-cfa2-4461-8a97-332ff5edf670) | 
|:--:| 
| *Logs of my local server* |

As you can see in the image below, I'm using `encore v1.41.1`.

![encore-version](https://github.com/user-attachments/assets/24677049-0f50-4c02-8833-2af5cc9bdd3d)

**Comment by fredr** [comment]
It looks like the connection string is correct, e.g you can use that to connect via `psql`, but when prisma tries to connect it fails for some reason. I'm currently looking into why and if we can find a way to fix that

**Comment by luisnquin** [comment]
Thanks, @fredr! It would be great if the [TS Database Documentation](https://encore.dev/docs/ts/primitives/databases) could also include references to ORMs that integrate seamlessly with this tool and its microservices architecture.

**Comment by fredr** [comment]
I've tested your migrate code, and for me it works if I delay execution like this:

```typescript
(async () => {
  await prismaMigrate(db.connectionString);
})();
```

Does that work for you as well?

**Comment by luisnquin** [comment]
@fredr yup, it's working! :)

![image](https://github.com/user-attachments/assets/11a8a367-62fa-4703-9d61-d8515a60bfaf)

**Comment by fredr** [comment]
Great,

in the next release of encore there is a fix for this (#1407) so you won't need to wrap it for long

**Comment by fredr** [comment]
If you update to `v1.41.4`, then your original code should work
